### PR TITLE
NIFI-2735: Add AggregateValues processor for aggregate operations on flow file groups

### PR DESCRIPTION
**Pull request by mattyb149**

### CODE REVIEW

**Comment by olegz** [comment]
Reviewing. . .

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AggregateValues.java [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+            .name("failure")
+            .description("If a FlowFile fails processing for any reason (for example, the FlowFile is not in a valid format or the specified "
+                    + "path does not exist), it will be routed to this relationship")
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private Set<Relationship> relationships;
+
+    static {
+        System.setProperty("javax.xml.xpath.XPathFactory:" + NamespaceConstant.OBJECT_MODEL_SAXON, "net.sf.saxon.xpath.XPathFactoryImpl");
```
Comment (olegz) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AggregateValues.java]: Given that NiFi operates in a single JVM instance do we have any other processor that may be doing the same thing yet with different value?

Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AggregateValues.java]: Not with a different value that I know of, I stole this from SplitXml or EvaluateXPath. Probably could move both to a separate place.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AggregateValues.java [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+        + "when a fragment/batch is complete. Supported MIME types include text/csv, text/plain, application/json, and application/xml. For text/csv and text/plain types, "
+        + "the flow file is expected to contain a single row of comma-delimited data (no headers). SplitText can be used for this purpose. For JSON files, SplitJson can be "
+        + "used to split the document (setting the expected attributes, etc.). For XML files, SplitXml can be used in a similar manner."
+        + "Note that all flow files with the same identifier must be routed to this processor, otherwise the count will need "
+        + "need to be decremented for each flow file in order to represent the number of flow files with the same identifier that will be routed to this processor. "
+        + "Once all fragments with the same identifier have been processed, a flow file will be transferred to the 'aggregate' relationship containing an attribute with the "
+        + "aggregated value")
+@Stateful(scopes = Scope.LOCAL, description = "The processor will temporarily keep state information for each fragment.identifier, including the current intermediateAggregate "
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AggregateValues.java]: This should be CLUSTER scope, will change.

**Comment by mattyb149** [comment]
Closing for now, rework is needed.

**Pull request closed without merging**